src/sandbox/components/CellEditor.svelte

```diff
@@ -1,23 +1,228 @@
+            found = true;
+            break;
+          }
+        }
+      }
+      if (found) break;
+      caretPosition += nodes[i].textContent.length;
+    }
+
+    if (!found) caretPosition = value.length;
+  }
+
+  function setCaretPosition() {
+    let caretNode = null;
+    let caretOffset = 0;
+
+    const highlight = document.querySelector("#highlight");
+    let currentPosition = 0;
+
+    if (highlight) {
+      const nodes = highlight.childNodes;
+
+      for (let i = 0; i < nodes.length; i++) {
+        const nodeLength = nodes[i].textContent.length;
+        currentPosition += nodeLength;
+
+        if (caretPosition <= currentPosition) {
+          caretNode = nodes[i].firstChild;
+          caretOffset = caretPosition - (currentPosition - nodeLength);
+          break;
+        }
+      }
+
+      if (!caretNode) {
+        caretNode = highlight.lastChild.lastChild;
+        caretOffset = caretNode.textContent.length;
+      }
+    }
+    else {
+      caretNode = node.lastChild;
+      caretOffset = caretPosition;
+    }
+    
+    document.getSelection().empty();
+    document.getSelection().setBaseAndExtent(caretNode, caretOffset, caretNode, caretOffset);
+  }
+
+  function handleInput() {
+    if (!node.textContent.length) return value = "";
+
+    onInput = true;
+    value = node.textContent;
+
+    getCaretPosition()
+
+    innerHTML = highlight(node.textContent);
+
+    tick().then(() => setCaretPosition());
+    autoSizeEditor();
+  }
+
+  async function autoSizeEditor() {
+    await tick();
+    if (node) {
+      node.style.removeProperty('width');
+      if (node.scrollWidth > node.clientWidth) {
+        node.style.width = node.scrollWidth + 5 + 'px';
+      }
+    }
+  }
+
+  onMount(() => {
+    innerHTML = highlight(value);
+
+    tick().then(() => {
+      let anchor = node;
+      let offset = 0;
+      if (value !== '') {
+        const str = document.querySelector("#highlight");
+        anchor = str? str.lastChild.lastChild : node.lastChild;
+        offset = str? str.lastChild.textContent.length : node.textContent.length;
+      }
+      document.getSelection().empty();
+      document.getSelection().setBaseAndExtent(anchor, offset, anchor, offset);
+    })
+
+    autoSizeEditor();
+  })
+
+  $: if (value && !onInput) {
+    innerHTML = highlight(value);
+    tick().then(() => setCaretPosition());
+    autoSizeEditor();
+  }
 </script>
 
-<textarea
+<div
   tabindex="-1"
+  id="editor"
   class="grid__editor grid__selection-border bg-white"
   class:dark:text-white={!transparent}
   class:dark:text-gray-800={transparent}
   class:dark:bg-slate-800={!transparent}
   bind:this={node}
-  bind:value
-  on:input
+  bind:innerHTML
+  contenteditable="true"
+  on:input={handleInput}
   on:keydown
-  autocomplete="off"
-  autocorrect="off"
+  on:paste={(e) => {
+    e.preventDefault();
+    let text = (e.originalEvent || e).clipboardData.getData('text/plain');
+    document.execCommand("insertHTML", false, text);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand
Deprecated: This feature is no longer recommended. 
No other option here?